specification/cloudhealth/CloudHealth.Management/main.tsp

```diff
@@ -0,0 +1,843 @@
+alias modelResourceNameRegex = "^[a-zA-Z][a-zA-Z0-9-]{1,42}[a-zA-Z0-9]$";
+
+@pattern(modelResourceNameRegex)
+@doc("String with regex validation to match the health model resource name requirements")
+scalar stringModelResourceName extends string;
+
+// Regex for all our proxy resource name validations. Up to 260 characters, alphanumeric and hyphens, must start and end with an alphanumeric character
+alias proxyResourceNameRegex = "^[a-zA-Z0-9][a-zA-Z0-9-]{1,258}[a-zA-Z0-9]$";
+
+@pattern(proxyResourceNameRegex)
+@doc("String with regex validation to match the proxy resource name requirements")
+scalar stringResourceName extends string;
+
+@maxLength(4096)
+@doc("String with max length validation of 4096 characters")
+scalar string4096 extends string;
+
+interface Operations extends Azure.ResourceManager.Operations {}
+
+@doc("A HealthModel resource")
+model HealthModel is TrackedResource<HealthModelProperties> {
+  @doc("Name of health model resource")
+  @key("healthModelName")
+  @path
+  @segment("healthmodels")
+  name: stringModelResourceName;
+
+  ...ManagedServiceIdentityProperty;
+}
+
+@doc("HealthModel properties")
+model HealthModelProperties {
+  @visibility(Lifecycle.Read)
+  @doc("The data plane endpoint for interacting with health data")
+  dataplaneEndpoint?: string;
+
+  @visibility(Lifecycle.Read)
+  @doc("The status of the last operation.")
+  provisioningState?: HealthModelProvisioningState;
+
+  @doc("Configure to automatically discover entities from a given scope, such as a Service Group. The discovered entities will be linked to the root entity of the health model.")
+  discovery?: ModelDiscoverySettings;
+}
+
+#suppress "@azure-tools/typespec-azure-core/documentation-required" "Self-explanatory"
+@doc("Health Model provisioning states")
+union HealthModelProvisioningState {
+  string,
+  ResourceProvisioningState, // include standard provisioning states
+  Creating: "Creating",
+  Deleting: "Deleting",
+}
+
+@doc("Settings for automatically discovering entities for the health model.")
+model ModelDiscoverySettings {
+  @doc("The scope from which entities should be automatically discovered. For example, the resource id of a Service Group.")
+  scope: string4096;
+
+  @doc("Whether to add all recommended signals to the discovered entities.")
+  addRecommendedSignals: DiscoveryRuleRecommendedSignalsBehavior;
+
+  @doc("Which Managed Identity of the health model to use for discovery. Defaults to SystemAssigned, if not set. Can be set to 'SystemAssigned' or to the resource id of a user-assigned managed identity which is linked to the health model.")
+  identity?: string4096;
+}
+
+@doc("A signal definition in a health model")
+@parentResource(HealthModel)
+model SignalDefinition is ProxyResource<SignalDefinitionProperties> {
+  @doc("Name of the signal definition. Must be unique within a health model.")
+  @key("signalDefinitionName")
+  @segment("signaldefinitions")
+  @path
+  name: stringResourceName;
```

Comment:
>stringResourceName [](http://example.com/codeflow?start=8&length=18)

Maintainability/usability: shouldn't this typed as something more like signalDefinitionName instead of something so generic sounding as stringResourceName?

Reply:
It probably doesn't matter, but I feel like if you are going to get into defining these types, this category of proxyResourceName is going to feel like it needs a clear definition, to your API users, so they can understand what it means, and yet also it doesn't feel like it should be a necessary concept to explain the API.

Reply:
sure, will change it. 

Just for my understanding: are these symbolic reference names being used / exposed really anywhere outside of the typespec file? I see it creates a type in the Swagger, but that piece also doesn't seem to be of interest to users?! Or does this bleed into the generated SDKs etc?

Reply:
change implemented.